### DISCUSSION TITLE
Unify DIP port names

### DISCUSSION THREAD

**Discussion by jotego** [mamedev/discussions#65] [General]
The name used in `PORT_START` and `PORT_DIPLOCATION` for DIP switches is not homogenous. It may be called `SWA` or `SW(A)` or `SW A` or `DIP-` or `DIPSW1` or `DSW-1` or `DSW0`... Whether it starts counting from `0`, `1` or `A` seems arbitrary too. These two names do not always agree with each other either. The bit mask used in the driver is not linked to the DIP location either and it may be in reverse order where DIP1 is mapped above DIP2 in a 16-bit word.

For documentation and for tools depending on MAME's XML output, having a homogeneous naming, such as `DIP1, DIP2...` each made of 8 bits, would be benefitial.

I am willing to make these changes now for some 50+ drivers and keep updating other drivers over time. Before spending time on this, is my PR likely to be accepted?

- Add PORT_DIPLOCATION where missing
- Follow an uniform name on all drivers, `DIP` plus an index starting at 1, increase every 8 bits. Except where the manual uses a letter or uses 4-bit switches
- Use `DSW` plus an index starting at 1 and increasing every 8 bits for PORT_START, regardless of manual. Order by bit mask used in the driver

In this way, the `PORT_DIPLOCATION` tries to document the board in an uniform way -but respectful with the manual- and `PORT_START` represents the emulation model of the hardware, but is uniform across drivers

**Comment by cuavas** [comment]
You’re looking at this from the wrong direction.  It isn’t abut making thing convenient in the XML, and the port names (in `PORT_START`) and switch location names (in `PORT_DIPLOCATION`) serve different purposes.  I can immediately identify numerous issues with what you’re proposing:
* First of all, a single I/O port may not correspond directly to a single group of DIP switches.  There are many cases where DIP switches and other inputs are read as a single logical port from the system’s perspective.  It’s also possible for a different parts of a single bank of DIP switches to be read via different logical ports.
* The I/O ports should be named in a way that makes sense.  Where possible, we used names used in schematics/service manuals, displayed in test modes, used in debugging symbols in the code, or similar.  There’s obviously no consistency in this across manufacturers.  If that isn’t possible, the port names should be something that makes sense to the developer.
* I/O port names have additional restrictions on them because they’re tags in the MAME sense of the word, and sometimes there are a large number of the that would be impractical to use without using object array finders.
* If possible, DIP switch names in `PORT_DIPLOCATION` should match the names displayed in test mode, used in manuals, silk screened on PCBs, etc.  The numbering needs to match the numbering in the DIP switches themselves – they may be wired MSB to LSB, LSB to MSB, or some other random order.  You shouldn’t just make it up for systems where we don’t have actual switch names and numbering available available.

You’re welcome to add `PORT_DIPLOCATION` for systems where test modes, manuals, photos showing PCB silkscreens, schematics, etc. showing the relevant information are available.  But please don’t go adding arbitrary `PORT_DIPLOCATION` where the information isn’t available – it will just confuse people using MAME to figure out how DIP switches work on their hardware.
